### DISCUSSION TITLE
How to extract network with openpnm

### DISCUSSION THREAD

**Discussion by Xiaozizhuo** [PMEAL/OpenPNM#2438] [Q&A]
Dear Professor jeff, I used other methods to generate a porous media model in MATLAB and extracted the network model. But at the same time I used OpenPNM to extract the network for this model to compare my reconstruction method is feasible. Can this function be done?
![6@ZI` 7P}U{IT{1BB7Q6~9O](https://user-images.githubusercontent.com/98071787/168805716-e2967526-594c-49b9-a71d-eea9268cb82b.png)

**Comment by jgostick** [comment]
can you rephrase your question?  Im not quite sure what you're asking.

**Reply by Xiaozizhuo** [reply]
Ok, I want to use OpenPNM to extract the network for the porous media model I generated, and then do a simple flow simulation based on the network. But I don't know how to open the model generated in MATLAB in OpenPNM

**Reply by jgostick** [reply]
So you have generated the above image in matlab AND performed an extraction in matlab as well, which is giving you the image on the right?  And you want to import this into OpenPNM so that you can run various simulations, correct?

If I'm correct in my interpretation above, then you'll need to output the data from Matlab into a format that OpenPNM can read.  We have several options in the ``openpnm.io`` module, include the ability to read "mat" files.  However, the data must be formated correctly.  The convention is that each pore and throat property is given a column, and named something like 'pore.radius' or 'throat.volume'.  You call need to have columns for the pore coordinates (pore.coords) and throat connections ('throat.conns').  Coordinates are in the form of an Np-by-3 array [x, y, z] coordiantes for each pore.  Connections are in the format of the sparse adjacency matrix in COO format.  This means that if pore i and j and connecte by throat 0, then the first row of the 'throat.conns' array would be [i, j].  

The best way to learn how to do this is to create a network in openpnm then 'reverse engineer' it by outputting it to the 'mat' file format, inspecting that file, then making your matlab output the same thing.

**Reply by Xiaozizhuo** [reply]
Yes professor, your method is useful thanks for your kind reply.
